pkg/codegen/schema.go

```diff
@@ -580,7 +580,16 @@ func GenFieldsFromProperties(props []Property) []string {
 			}
 		}
 
-		field += fmt.Sprintf("    %s %s", goFieldName, p.GoTypeDef())
+		if p.Schema.AdditionalTypes == nil {
+			field += fmt.Sprintf("    %s %s", goFieldName, p.GoTypeDef())
+		} else {
+			for _, td := range p.Schema.AdditionalTypes {
+				if p.Schema.TypeDecl() == td.TypeName {
+					out, _ := GenerateGoSchema(p.Schema.OAPISchema.NewRef(), nil)
+					field += fmt.Sprintf("    %s %s", goFieldName, out.GoType)
+				}
+			}
+		}
```

Comment:
I'm working on the same problem in a very different way in #654. I have to admit, this is an incredibly elegant and tight solution you have here if it works. Does it work on enums?

Reply:
It doesn't appear to :grin: We'll need to use a combination of the two, methinks, to get this working